src/plugins/orangehrmRecruitmentPlugin/Api/CandidateAPI.php

```diff
@@ -0,0 +1,639 @@
+            )
+        );
+        $candidate->setDateOfApplication(
+            $this->getRequestParams()->getDateTimeOrNull(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_DATE_OF_APPLICATION,
+                null,
+                $this->getDateTimeHelper()->getNow()
+            )
+        );
+        $candidate->setModeOfApplication(
+            $this->getRequestParams()->getInt(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_MODE_OF_APPLICATION
+            )
+        );
+        $candidate->setStatus(
+            $this->getRequestParams()->getInt(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_STATUS
+            )
+        );
+        $candidate->setConsentToKeepData(
+            $this->getRequestParams()->getBooleanOrNull(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_CONSENT_TO_KEEP_DATA,
+                false
+            )
+        );
+        $candidate->getDecorator()->setAddedPersonById(
+            $this->getAuthUser()->getEmpNumber()
+        );
+    }
+
+    /**
+     * @param CandidateVacancy $candidateVacancy
+     * @param int $candidateId
+     */
+    private function setCandidateVacancy(CandidateVacancy $candidateVacancy, int $candidateId, string $status)
+    {
+        $vacancyId = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_VACANCY_ID
+        );
+        $candidateVacancy->getDecorator()->setCandidate($candidateId);
+        $candidateVacancy->getDecorator()->setVacancy($vacancyId);
+        $candidateVacancy->setStatus($status);
+        $candidateVacancy->setAppliedDate($this->getDateTimeHelper()->getNow());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(...$this->getCommonBodyValidationRules());
+    }
+
+    /**
+     * @return array
+     */
+    private function getCommonBodyValidationRules(): array
+    {
+        return [
+            new ParamRule(
+                self::PARAMETER_FIRST_NAME,
+                new Rule(Rules::STRING_TYPE)
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_MIDDLE_NAME,
+                    new Rule(Rules::STRING_TYPE)
+                ),
+                true
+            ),
+            new ParamRule(
+                self::PARAMETER_LAST_NAME,
+                new Rule(Rules::STRING_TYPE)
+            ),
+            new ParamRule(
+                self::PARAMETER_EMAIL,
+                new Rule(Rules::EMAIL)
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_CONTACT_NUMBER,
+                    new Rule(Rules::PHONE)
```

Comment:
@devishke-orange please check this

Reply:
yes should be fine. candidate screen in 4.x has standard message when invalid phone number is entered. this rule checks for that same format I think it's okay